worker/lease/store.go

```diff
@@ -38,6 +41,11 @@ type Store struct {
 	db     *sql.DB
 	logger StoreLogger
 
+	// ctx is passed to *all* database interactions that can accept a context.
+	// It must be cancelled in the worker's Kill method so no such operations
+	// can block a worker shutdown.
+	ctx context.Context
```

Comment:
I have not undertaken that in this patch, but it could be done:
- Change the `boundManager` methods (and specifications) to accept a context; these will come from API calls.
- Call `tomb.Context` with those as arguments, and hand the new child context down.
- Pass that context's `Done` channel as the stop channel to all the ops that thread requests into the worker loop.